Components/Form.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+
+function Form() {
+  const [todo, setTodo] = useState("");
+  const [todoList, setTodoList] = useState([]);
+  
+  const handleChange = (e) => {
+    setTodo(e.target.value);
+  };
+
+  const addTodo = () => {
+    if (todo != "") {
+      const todoInsert = {
+        id: Math.floor(Math.random() * 10000), //Generate random id from 0 to 10000
+        value: todo,
+        isCompleted: false,
+      };
+
+      setTodoList([...todoList, todoInsert]); //... operator "spreads out the own enumerable properties as properties of the element you're creating" stack overflow
+      setTodo(" ");
+    }
+  };
+
+  const deleteTodo = (e, id) => {
+    e.preventDefault();
+    setTodoList(todoList.filter((t) => t.id != id));
+  };
+
+  const completeTodo = id => {
+   setTodoList(todoList.map(item => {
+       if(item.id == id){
+        item.isCompleted = !item.isCompleted   
+       }
+       return item  
+     })
+   )
+  }
+
+  return (
+    <div>
+      <label htmlFor="newItem">Nuovo item</label>
+      <input
+        type="text"
+        name="newItem"
+        id="newItem"
+        onChange={(e) => handleChange(e)}
```

Comment:
I didn't spot this one at first, but this input is "uncontrolled", meaning that it is not controlled by the `todo` state, so for example when you reset it nothing changes.
```suggestion
        onChange={(e) => handleChange(e)}
        value={todo}
```